tests/test_hydrology.py

```diff
@@ -151,3 +151,55 @@ def test_simple(self, q_series):
         out = xci.low_flow_frequency(q, 0.2, freq="YS")
 
         np.testing.assert_array_equal(out, [20, 0])
+
+
+class TestAntecedentPrecipitationIndex:
+    def test_simple(self, pr_series):
+        a = np.ones(50) * 10
+        a[15:20] = 20
+        a[35:40] = 0
+        pr = pr_series(a)
+        window = 7
+        out = xci.antecedent_precipitation_index(pr, window=window, p_exp=0.935)
+        np.testing.assert_array_equal(
+            out.max().round(2),
+            [
+                101.65,
+            ],
+        )
+        np.testing.assert_array_equal(
+            out.min().round(2),
+            [
+                13.83,
+            ],
+        )
+
+    def test_nan_present(self, pr_series):
+        a = np.ones(50) * 10
+        a[25] = np.nan
+        pr = pr_series(a)
+        window = 7
+        out = xci.antecedent_precipitation_index(pr, window=window, p_exp=0.935)
+        np.testing.assert_array_equal(out[25], [np.nan])
+
+    def test_nan_start_window(self, pr_series):
+        a = np.ones(50) * 10
+        pr = pr_series(a)
+        window = 7
+        out = xci.antecedent_precipitation_index(pr, window=window, p_exp=0.935)
+        np.testing.assert_array_equal(out[: window - 1], [np.nan for _ in range(window - 1)])
+
+    def test_manual_calc(self, pr_series):
+        a = np.ones(10) * 10
+        pr = pr_series(a)
+        window = 7
+        p_exp = 0.935
+        out = xci.antecedent_precipitation_index(pr, window=window, p_exp=p_exp)
+
+        out_manual = np.zeros(out.shape) * np.nan
+        for idx in range(pr.shape[0] - window + 1):
+            idxend = window + idx
+            weights = list(reversed([p_exp ** (ii + 1 - 1) for ii in range(window)]))
+            weighted_sum = (pr[idx:idxend] * weights).sum()
+            out_manual[idxend - 1] = weighted_sum
+        np.testing.assert_array_equal([round(x, 7) for x in out.values], [round(x, 7) for x in out_manual])
```

Comment:
```suggestion
        np.testing.assert_allclose(out, out_manual, atol=1e-7)
```
Let numpy do the tolerance bit.  Also, FYI : `out.round(7)` is available on DataArrays.